client/lxd_containers.go

```diff
@@ -1403,3 +1403,69 @@ func (r *ProtocolLXD) DeleteContainerTemplateFile(name string, templateName stri
 	_, _, err := r.query("DELETE", fmt.Sprintf("/containers/%s/metadata/templates?path=%s", name, templateName), nil, "")
 	return err
 }
+
+// ConsoleContainer requests that LXD attaches to a console or queries the
+// console log of the container.
+func (r *ProtocolLXD) ConsoleContainer(containerName string, console api.ContainerConsolePost, args *ContainerConsoleArgs) (*Operation, error) {
+	if !r.HasExtension("console_attach") {
+		return nil, fmt.Errorf("The server is missing the required \"console_attach\" API extension")
+	}
+
+	// Send the request
+	op, _, err := r.queryOperation("POST", fmt.Sprintf("/containers/%s/console", containerName), console, "")
+	if err != nil {
+		return nil, err
+	}
+
+	if args == nil {
+		return nil, fmt.Errorf("Please pass stdin, stdout, and stderr")
+	}
+
+	if args.Stdin == nil {
+		return nil, fmt.Errorf("Please pass stdin")
+	}
+
+	if args.Stdout == nil {
+		return nil, fmt.Errorf("Please pass stdout")
+	}
+
+	if args.Stderr == nil {
+		return nil, fmt.Errorf("Please pass stderr")
+	}
```

Comment:
If we got with my suggestion above, this should simplify those checks quite a bit to the point where we can just do:

    if args == nil || args.Terminal == nil {